diff --git a/src/wolfsftp.c b/src/wolfsftp.c
index e32d09d..d2e2d37 100644
--- a/src/wolfsftp.c
+++ b/src/wolfsftp.c
@@ -2565,7 +2565,7 @@ static int SFTP_CreateLongName(WS_SFTPNAME* name)
     char perm[11];
     int linkCount = 1; /* @TODO set to correct value */
     char date[WS_DATE_SIZE + 1]; /* +1 for null terminator */
-    struct tm* localTime = NULL;
+    struct tm localTime;
     int i;
     WS_SFTP_FILEATRB* atr;
 #endif
@@ -2579,11 +2579,8 @@ static int SFTP_CreateLongName(WS_SFTPNAME* name)
     atr = &name->atrb;
 
     /* get date as calendar date */
-    localTime = XGMTIME((const time_t*)&atr->mtime, &localTime);
-    if (localTime == NULL) {
-        return WS_MEMORY_E;
-    }
-    getDate(date, sizeof(date), localTime);
+    XGMTIME((const time_t*)&atr->mtime, &localTime);
+    getDate(date, sizeof(date), &localTime);
     totalSz += WS_DATE_SIZE;
 
     /* set permissions */
@@ -3439,7 +3436,7 @@ int wolfSSH_SFTP_RecvWrite(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
             return WS_BUFFER_E;
         }
 
-        ret = WPWRITE(ssh->fs, fd, data + idx, sz, ofst);
+        ret = WPWRITE(ssh->fs, fd, data + idx, sz, (off_t*)ofst);
         if (ret < 0) {
     #if defined(WOLFSSL_NUCLEUS) && defined(DEBUG_WOLFSSH)
             if (ret == NUF_NOSPC) {
@@ -3626,7 +3623,7 @@ int wolfSSH_SFTP_RecvRead(WOLFSSH* ssh, int reqId, byte* data, word32 maxSz)
         return WS_MEMORY_E;
     }
 
-    ret = WPREAD(ssh->fs, fd, out + UINT32_SZ + WOLFSSH_SFTP_HEADER, sz, ofst);
+    ret = WPREAD(ssh->fs, fd, out + UINT32_SZ + WOLFSSH_SFTP_HEADER, sz, (off_t*)ofst);
     if (ret < 0 || (word32)ret > sz) {
         WLOG(WS_LOG_SFTP, "Error reading from file");
         res  = err;
